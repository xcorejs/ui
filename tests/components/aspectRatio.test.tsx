import 'jest-styled-components';

import { AspectRatio } from '@xcorejs/ui';
import React from 'react';
import renderer from 'react-test-renderer';

test('AspectRatio component', () => {
  const component = renderer.create(
    <>
      <AspectRatio ratio={9 / 16}>
        16:9
      </AspectRatio>
      <AspectRatio ratio={3 / 4}>
        4:3
      </AspectRatio>
    </>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
