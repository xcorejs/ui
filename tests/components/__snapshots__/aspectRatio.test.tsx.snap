// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AspectRatio component 1`] = `
Array [
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  top: 0;
}

.c0 {
  width: 100%;
  height: 0;
  padding-bottom: 56.25%;
  position: relative;
}

.c0 > div {
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.c0 img {
  object-fit: cover;
}

<div
    className="c0"
    height={0}
    width="100%"
  >
    <div
      className="c1"
    >
      16:9
    </div>
  </div>,
  .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  top: 0;
}

.c0 {
  width: 100%;
  height: 0;
  padding-bottom: 75%;
  position: relative;
}

.c0 > div {
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.c0 img {
  object-fit: cover;
}

<div
    className="c0"
    height={0}
    width="100%"
  >
    <div
      className="c1"
    >
      4:3
    </div>
  </div>,
]
`;
